File: VB/E1554.Module/Master.vb
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation
Imports DevExpress.Data.Filtering

Namespace E1554.Module
	<DefaultClassOptions>
	Public Class Master
		Inherits BaseObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _MasterName As String
		Public Property MasterName() As String
			Get
				Return _MasterName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("MasterName", _MasterName, value)
			End Set
		End Property
		<Association("Master-Details")>
		Public ReadOnly Property Details() As XPCollection(Of Detail)
			Get
				Return GetCollection(Of Detail)("Details")
			End Get
		End Property
	End Class

End Namespace

File: CS/EFCore/DialogBeforeListViewEF/DialogBeforeListViewEF.Module/BusinessObjects/Master.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl.EF;
using System.Collections.ObjectModel;

namespace E1554.Module {
    [DefaultClassOptions]
    public class Master : BaseObject {
        public virtual string MasterName { get; set; }
        public virtual IList<Detail> Details { get; set; } = new ObservableCollection<Detail>();
    }

}